src/reporter/health.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+
+def check_crate():
+    """
+    crateDB is the default backend of QuantumLeap, so it is required by
+    default.
+    """
+    from translators.crate import CrateTranslatorInstance
+    with CrateTranslatorInstance() as trans:
+        crate_health = trans.get_health()
+        return crate_health
+
+
+def check_geocache():
+    """
+    Geocache is relevant only when geocoding usage is enabled.
+    """
+    use_geocoding = os.environ.get('USE_GEOCODING', False)
+    if not use_geocoding:
+        return {'status': 'pass'}
+
+    from geocoding.geocache import GeoCodingCache
+    host = os.environ.get('REDIS_HOST', None)
+    port = os.environ.get('REDIS_PORT', 6379)
+    gc = GeoCodingCache(host, port)
+    health = gc.get_health()
+    return health
+
+
+def check_geocoder():
+    """
+    Geocoder is relevant only when geocoding usage is enabled.
+    """
+    use_geocoding = os.environ.get('USE_GEOCODING', False)
+    if not use_geocoding:
+        return {'status': 'pass'}
+
+    from geocoding import geocoding
+    health = geocoding.get_health()
+    return health
+
+
+def _get_http_code(res):
+    if res['status'] == 'pass':
+        code = 200
+    elif res['status'] == 'warn':
+        code = 207
+    else:
+        assert res['status'] == 'fail'
+        code = 424
```

Comment:
shouldn't be 503? (service not available)

Reply:
I'm not sure really, it's kind of a paradox. My reasoning was: I would expect from the server a 503 if the `/health` service was down so I can't know what's the health. But if it's working and telling you something's wrong with the dependencies, then we could use [424 Failed Dependency](https://httpstatuses.com/424)

Reply:
what's elk giving back?

Reply:
Well, not sure, because it depends on the error type. I see you could get http 200 even with some red status (e.g. with one broken shard,  in which case it'd still return partial results).

Reply:
Since there's no standard, I thought of following this draft: https://inadarei.github.io/rfc-healthcheck/
which suggests that fail status should return codes in the 4XX/5XX range.

Reply:
wait a minute, if the service is partially working, you don't have a fail.

Reply:
I took it from here: https://www.elastic.co/guide/en/elasticsearch/reference/current/_cluster_health.html

Reply:
OK, let's use 503. On a second thought, this looks more reasonable/common.